gui/create-item.lua

```diff
@@ -184,11 +211,176 @@ local function getCreatureRaceAndCaste(caste)
+ local layerMat
+ local tissueID
+ 
+ if not generic then -- if we have a specified body part and layer, figure all the stuff out about that
+ -- store the tissue id of the specific layer we selected
+  tissueID = tonumber(creatorBody.body_parts[bodypart].layers[partlayer].tissue_id)
+  layerMat = creatorRaceRaw.tissue[tissueID].tissue_material_str[1]
+  layerName = creatorBody.body_parts[bodypart].layers[partlayer].layer_name
+ else -- otherwise, figure out the mat name from the dual-use partlayer argument
+  layerMat = creatorRaceRaw.material[partlayer].id
+  layerName = layerMat
+ end
+
+ -- default is MEAT, so if anything else fails to change it to something else, we know that the body layer is a meat item
+ local item_type = "MEAT"
+ -- get race name and layer name, both for finding the item material, and the latter for determining the corpsepiece flags to set
+ local raceName = string.upper(creatorRaceRaw.creature_id)
+ -- every key is a valid non-hair corpsepiece, so if we try to index a key that's not on the table, we don't have a non-hair corpsepiece
+ local corpseTable = {BONE = "BONE", SKIN = "SKIN", CARTILAGE = "CARTILAGE", TOOTH = "TOOTH", NERVE = "NERVE", NAIL = "NAIL", HORN = "HORN", HOOF = "HOOF"}
+ -- we do the same as above but with hair
+ local hairTable = {HAIR = "HAIR", EYEBROW = "EYEBROW", EYELASH = "EYELASH", MOUSTACHE = "MOUSTACHE", CHIN_WHISKERS = "CHIN_WHISKERS", SIDEBURNS = "SIDEBURNS"}
+ -- if the layer is fat, spawn a glob of fat and DON'T check for other layer types
+ if layerName == "FAT" then
+  item_type = "GLOB"
+ elseif corpseTable[layerName] or hairTable[layerName] then -- check if hair
+  -- farlayer = 5
+  -- layerName = "HAIR" -- we then simplify every hair-tissue into just "HAIR"
+  item_type = "CORPSEPIECE"
+ end
+ 
+ local itemType = dfhack.items.findType(item_type..":NONE")
+ local itemSubtype = dfhack.items.findSubtype(item_type..":NONE")
+ local material = "CREATURE_MAT:"..raceName..":"..layerMat
+ local materialInfo = dfhack.matinfo.find(material)
+ 
+ local item_id = dfhack.items.createItem(itemType, itemSubtype, materialInfo['type'], materialInfo.index, creator)
+ local item = df.item.find(item_id)
+ 
+ item:setQuality(quality)
+ 
+ local isCorpsePiece = false
+ 
+ -- if the item type is a corpsepiece, we know we have one, and then go on to set the appropriate flags
+ if item_type == "CORPSEPIECE" then
+	 if layerName == "BONE" then -- check if bones
+		item.corpse_flags.bone = true
+		item.material_amount.Bone = 1
+	 elseif layerName == "SKIN" then -- check if skin/leather
+		item.corpse_flags.leather = true
+		item.material_amount.Leather = 1
+	 -- elseif layerName == "CARTILAGE" then -- check if cartilage (NO SPECIAL FLAGS)
+	 elseif layerName == "HAIR" then -- check if hair (simplified from before)
+		item.corpse_flags.hair_wool = true
+		item.material_amount.HairWool = 1
+		if materialInfo.material.flags.YARN then
+			item.corpse_flags.yarn = true
+			item.material_amount.Yarn = 1
+		end
```

Comment:
to verify, if it's yarn, then both HairWool *and* Yarn get set?

Reply:
![Screenshot_7802](https://user-images.githubusercontent.com/78707934/215354914-79b9c6ef-d311-4d82-9e27-2dace5e3360d.png)
ye.